diagnostic.md

```diff
@@ -6,70 +6,73 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What are two ways of telling rails where to route a GET request to '/books/4'?
 
 ```rb
-// your response here
+esources :books, only: [:index, :show]
+or
+resources :books, except: [:new, :edit]
 ```
 
 In what file does the above code go?
 
 ```md
-// your response here
+config/routes.rb
 ```
 
 In your own words, what does the command `bundle exec rake db:migrate` do?
 
 ```md
-// your response here
+we are calling the bundler to execute any/all scripts in db/migrate directory
 ```
 
 What is the difference between `rails g scaffold things` and
 `rails g controller things`?
 
 ```md
-// your response here
+you are generateing a model and generating a controller, respectively
+the scaffold will create a model, a controller for that model, as well as default views
 ```
 
 What is the job of the `serializer`?
 
 ```md
-// your response here
+the serializer is used to serve up JSON
```

Comment:
The serializer allows you to filter certain data out of your response. The data stil exists, it's just not shown in the response to users.